CAIPs/caip-x.md

```diff
@@ -0,0 +1,326 @@
+      "eip155:137": {
+        "methods": ["eth_sendTransaction"],
+        "notifications": ["chainChanged"]
+      }
+    }
+  }
+}
+```
+
+**Explanation:**
+
+- The `provider_augment` method adds the specified scopes to the current session's permissions. If the scope already exists, the new methods and notifications are merged with the existing ones.
+
+#### `provider_revoke`
+
+Revokes permissions from an existing session.
+
+**Parameters:**
+
+- `sessionId` (string, optional): The session identifier.
+- `scopes` (object, required): An object containing the scopes to be revoked, formatted according to CAIP-217.
+
+**Initial Session Scopes:**
+
+```json
+{
+  "eip155:1": {
+    "methods": ["eth_signTransaction", "eth_sendTransaction"],
+    "notifications": ["accountsChanged", "chainChanged"]
+  },
+  "eip155:137": {
+    "methods": ["eth_sendTransaction"],
+    "notifications": ["chainChanged"]
+  }
+}
+```
+
+**Example Request:**
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "provider_revoke",
+  "params": {
+    "scopes": {
+      "eip155:1": {
+        "methods": ["eth_sendTransaction"],
+        "notifications": ["chainChanged"]
+      }
+    }
+  }
+}
+```
+
+**Example Response:**
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "success": true,
+    "updatedScopes": {
+      "eip155:1": {
+        "methods": ["eth_signTransaction"],
+        "notifications": ["accountsChanged"]
+      },
+      "eip155:137": {
+        "methods": ["eth_sendTransaction"],
+        "notifications": ["chainChanged"]
+      }
+    }
+  }
+}
+```
+
+**Explanation:**
+
+- The `provider_revoke` method removes the specified methods and notifications from the current session's permissions. If the scope no longer has any methods or notifications after revocation, it is removed from the session.
+
+#### `provider_getSession`
+
+Retrieves the current permissions of an existing session.
+
+**Parameters:**
+
+- `sessionId` (string, optional): The session identifier.
+
+**Initial Session Scopes:**
+
+```json
+{
+  "eip155:1": {
+    "methods": ["eth_signTransaction"],
+    "notifications": ["accountsChanged"]
+  },
+  "eip155:137": {
+    "methods": ["eth_sendTransaction"],
+    "notifications": ["chainChanged"]
+  }
+}
+```
+
+**Example Request:**
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "provider_getSession",
+  "params": {}
+}
+```
+
+**Example Response:**
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "scopes": {
+      "eip155:1": {
+        "methods": ["eth_signTransaction"],
+        "notifications": ["accountsChanged"]
+      },
+      "eip155:137": {
+        "methods": ["eth_sendTransaction"],
+        "notifications": ["chainChanged"]
+      }
+    }
+  }
+}
+```
+
+**Explanation:**
+
+- The `provider_getSession` method returns the current permissions for the session. It lists all scopes along with their methods and notifications.
+
+### Events
+
+#### `providerAuthorizationUpdated`
```

Comment:
thoughts on `providerAuthorizationChanged`? matches the existing ethereum notifications

Reply:
That works!

Reply:
changed here: https://github.com/ChainAgnostic/CAIPs/pull/285/commits/51721fd888285fe5bba9bd3ca6d98929c73c5772